src/Paprika/Merkle/ComputeMerkleBehavior.cs

```diff
@@ -546,16 +543,24 @@ private void EncodeBranch(scoped in Key key, scoped in ComputeContext ctx, scope
                     if (memoize)
                     {
                         memoizedUpdated = true;
-                        memo.Set(keccakOrRlp, i);
+
+                        if (memo.Exists(i, branch.Children))
+                        {
+                            memo.Set(keccakOrRlp, i, branch.Children);
+                        }
+                        else if (keccakOrRlp.DataType == KeccakOrRlp.Type.Keccak)
+                        {
+                            memo = RlpMemo.Insert(memo, i, branch.Children, keccakOrRlp.Span, rlpMemoization);
```

Comment:
What happens if encoded data in are not a `Keccak`? Do we store a `Keccak.Zero` then? If not, how does it distinguish between existence of a keccak or not? Doesn't it mean that RlpMemo requires a similar NibbleSet to store bits of what was remembered or not?